bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -86,10 +137,21 @@
     }
 
     @Activate
-    public void activate() {
+    protected void activate() {
+        start();
+    }
+
+    private void start() {
         for (InetAddress address : getAllInetAddresses()) {
             createJmDNSByAddress(address);
         }
+        for (ServiceDescription description : activeServices) {
+            try {
+                registerService(description);
```

Comment:
Didn't you rather intend to call `registerServiceInternal` here?

Reply:
Yes, of course.

Reply:
Unregistering the services is not needed, if the MDNS instances the services were registered to are terminated.